### PR TITLE
add build recipe

### LINKED ISSUE #69: Create conda recipe
Create an Anaconda recipe for uploading the library to an Anaconda channel.

### PR DESCRIPTION
**Pull request by tgiroux**
resolves #69 

ran conda-build with no errors.
ran ```conda install <tar.bz2>``` with no errors.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+build:
+  string: {{ environ.get("GIT_DESCRIBE_HASH", "") }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}  # [linux]
+    - vc 14  # [win]
+    - cmake >=3.10
+  run:
+    - vc  # [win]
```
Comment (Kelvinrr) [inline recipe/meta.yaml:16]: you basically need to add all the dependencies in environment.yml here. I know it's redundant but I am not 100% sure of a better way other than some custom script which is more of a hassle.

File: recipe/build.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+mkdir build && cd build
+cmake ..
```
Comment (Kelvinrr) [inline recipe/build.sh:3]: add `-DCMAKE_INSTALL_PREFIX=$PREFIX` to the cmake command

File: recipe/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  git_url: https://github.com/USGS-Astrogeology/SugarSpice.git
+build:
+  string: {{ environ.get("GIT_DESCRIBE_HASH", "") }}
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}  # [linux]
+    - vc 14  # [win]
+    - cmake >=3.10
+#    - cspice-cmake
```
Comment (tgiroux) [inline recipe/meta.yaml:15]: conda-build fails if this package is included. unable to find cspice-cmake.

Comment (tgiroux) [reply recipe/meta.yaml]: turns out the environment.yml was creating an environment without tudat-team in the channels ?? resolved by moving tudat-team to the bottom of the ```channels``` list in env yml

File: recipe/meta.yaml (hunk 1 above)
Comment (tgiroux) [reply recipe/meta.yaml:22]: new commits addressed this. I left out the dependencies for building documentation. ```sphinxcontrib.gist``` is only available as a pip package so it was causing failure

File: recipe/meta.yaml (hunk 3 above)
Comment (Kelvinrr) [reply recipe/meta.yaml]: probably a channel thing. Add the channels as `-c` flags. 

the channels being: 

```
channels:
  - tudat-team
  - defaults
  - conda-forge
  - usgs-astrogeology
```
so do like, `-c tudat-team -c defaults -c conda-forge -c astrogeology`. That being said, you can remove cspice from run and add it only to build as it's a private dependency.

File: recipe/build.sh [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+mkdir build && cd build
+cmake .. -DCMAKE_INSTALL_PREFIX=$PREFIX
```
Comment (Kelvinrr) [inline recipe/build.sh:3]: -c flags should go here btw

File: recipe/meta.yaml (hunk 3 above)
Comment (tgiroux) [reply recipe/meta.yaml]: Accomplished by running ```conda-build . -c tudat-team -c defaults -c conda-forge -c usgs-astrogeology```
Couldn't get it working any other way. I added a readme to recipe/ to point out that channels must be specified.

An open issue about specifying channels for a recipe
https://github.com/conda/conda-build/issues/532

**Comment by Kelvinrr** [comment]
it built locally, so this is good.

**Pull request merged**